homeassistant/components/venstar/climate.py

```diff
@@ -266,20 +271,20 @@ def _set_operation_mode(self, operation_mode):
     def set_temperature(self, **kwargs):
         """Set a new target temperature."""
         set_temp = True
-        operation_mode = kwargs.get(ATTR_HVAC_MODE, self._client.mode)
+        operation_mode = kwargs.get(ATTR_HVAC_MODE)
         temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
         temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
         temperature = kwargs.get(ATTR_TEMPERATURE)
 
-        if operation_mode != self._client.mode:
-            set_temp = self._set_operation_mode(operation_mode)
+        if self._mode_map.get(operation_mode) != self._client.mode:
+            set_temp = self._set_operation_mode(self._mode_map.get(operation_mode))
 
         if set_temp:
-            if operation_mode == self._client.MODE_HEAT:
+            if operation_mode == HVAC_MODE_HEAT:
```

Comment:
Here we should check against the existing client mode instead, ie either the new mode that we just set, or if we didn't change mode, the existing client mode.

Reply:
Why? I'm not sure I understand the reasoning here

Reply:
We want to set the temp set points according to the current client mode of the device, right? If the user didn't pass an operation mode argument we will not set the correct set point with the current code. We need to change to check the mode of the device instead.

Reply:
Ah got it. So instead of comparing `operation_mode` to `HVAC_MODE_HEAT`, we'd compare `self._client.mode` to `self._client.MODE_HEAT` right?

Reply:
Yes! Has `self._client.mode` had time to update correctly if we asked the libraryt to change it in the call above?

Reply:
No, if `operation_mode` is `None` then it won't ever pass the conditional check to change the operation mode.

 If `operation_mode` is not `None` and `set_temp` is `True` after we try to change the operation mode, then we can safely say that `operation_mode == self._mode_map.get(operation_mode)`

If `operation_mode` is `None` we should have no reason to believe that our value of `self._client.mode` is out of date because we haven't done anything with it. So we could safely compare `self._client.mode` to `self._client.(MODE_HEAT|MODE_COOL|MODE_AUTO)`

Or if we wanted to ensure we had the most up to date value for `self._client.mode` we'd have to either wait for another `update()` cycle or manually call `self._client.update_info()` before the comparison.

The 3rd option seems best, but runs the risk of having old data since the values are cached locally until you call `self._client.update_info()` again

Reply:
Can we check against `self._mode_map.get(operation_mode)` if that is not None or `self._client.mode`?

That seems simplest.